api/admin/controller.py

```diff
@@ -2886,3 +2872,98 @@ def _decrypt_shared_secret(self, encryptor, shared_secret):
                 _("Could not decrypt shared secret %s") % shared_secret
             )
         return shared_secret
+
+    def _register_library(self, catalog_url, library, integration,
+                          do_get=HTTP.debuggable_get, do_post=HTTP.debuggable_post, key=None):
+        """Attempt to register a library with an external service,
+        such as a library registry or a shared collection on another
+        circulation manager.
+
+        Note: this method does a commit in order to set a public
+        key for the external service to request.
+        """
+        response = do_get(catalog_url)
+        if isinstance(response, ProblemDetail):
+            return response
+        type = response.headers.get("Content-Type")
+        if type and type.startswith('application/opds+json'):
+            # This is an OPDS 2 catalog.
+            catalog = json.loads(response.content)
+            links = catalog.get("links", [])
+            vendor_id = catalog.get("metadata", {}).get("adobe_vendor_id")
```

Comment:
It seems odd to do this in general rather than only when registering with the library registry, but I don't think it will be a problem.

Reply:
Yeah, I agree.